test/test_testing.py

```diff
@@ -5485,3 +5488,199 @@ def test_bad_results(self):
         self._action_results["A"] = "foo"
         with self.assertRaises(ValueError):
             self.harness.run_action("results")
+
+
+class TestNotify(unittest.TestCase):
+    def test_notify_basics(self):
+        harness = ops.testing.Harness(ContainerEventCharm, meta="""
+            name: notifier
+            containers:
+              foo:
+                resource: foo-image
+              bar:
+                resource: foo-image
+        """)
+        self.addCleanup(harness.cleanup)
+        harness.begin()
+        harness.charm.observe_container_events('foo')
+        harness.charm.observe_container_events('bar')
+
+        id1a = harness.pebble_notify('foo', 'example.com/n1')
+        id1b = harness.pebble_notify('foo', 'example.com/n1')
+        id2 = harness.pebble_notify('foo', 'foo.com/n2')
+        id3 = harness.pebble_notify('bar', 'example.com/n1')
+
+        self.assertIsInstance(id1a, str)
+        self.assertNotEqual(id1a, '')
+        self.assertEqual(id1a, id1b)
+
+        self.assertIsInstance(id2, str)
+        self.assertNotEqual(id2, '')
+        self.assertNotEqual(id2, id1a)
+
+        self.assertIsInstance(id3, str)
+        self.assertNotEqual(id3, '')
+        self.assertNotEqual(id3, id2)
```

Comment:
is it woeth having an id1c after you have done other events, and show that it isnt just deduping if the event is the previous event?
either that or move the id1b to after 2 and 3 to get that effect.

Reply:
Moved -- thanks.